File: pytools/examples/csharp/client/Protocol/AckTestSendOk.cs
using System.Collections;
using System.Collections.Generic;


public class AckTestSendOk
{
	private byte _id_u8;
	private MsgRoleBase _role_base;
	private List<float> _id_f32 = new List<float>();
	private byte id_op_u8_flag;
	private byte _id_op_u8;
	private byte op_role_base_flag;
	private MsgRoleBase _op_role_base;


	public AckTestSendOk(Packet packet)
	{
		this._id_u8 = packet.ReadByte();
		this._role_base = new MsgRoleBase(packet);
		this._id_f32 = new List<float>();
		ushort id_f32_count = packet.ReadUshort();
		for (ushort i = 0; i < id_f32_count; i++)
		{
			this._id_f32.Add(packet.ReadFloat());
		}
		this. id_op_u8_flag = packet.ReadByte();
		if (this.id_op_u8_flag == 1)
		{
			this._id_op_u8 = packet.ReadByte();
		}
		this. op_role_base_flag = packet.ReadByte();
		if (this.op_role_base_flag == 1)
		{
			this._op_role_base = new MsgRoleBase(packet);
		}
	}


	public byte id_u8
	{
		get { return this._id_u8; }
		set { this._id_u8 = value; }
	}

	public MsgRoleBase role_base
	{
		get { return this._role_base; }
		set { this._role_base = value; }
	}

	public List<float> id_f32
	{
		get { return this._id_f32; }
		set { this._id_f32 = value; }
	}

	public byte id_op_u8
	{
		get { return this._id_op_u8; }
		set { this.id_op_u8_flag = 1; this._id_op_u8 = value; }
	}

	public MsgRoleBase op_role_base
	{
		get { return this._op_role_base; }
		set { this.op_role_base_flag = 1; this._op_role_base = value; }
	}

}

File: pytools/data/data.code.golang.client/ack_test_send_ok.go
package proto

import (
	"packet"
)

type AckTestSendOk struct {
	idU8                     uint8
	roleBase                 *MsgRoleBase
	idF32                    []float32
	idOpU8Flag               uint8
	idOpU8                   uint8
	opRoleBaseFlag           uint8
	opRoleBase               *MsgRoleBase
}

func AckTestSendOkDecode(pack *packet.Packet) *AckTestSendOk {
	ackTestSendOk := &AckTestSendOk{}

	ackTestSendOk.idU8 = pack.ReadUint8()
	ackTestSendOk.roleBase = MsgRoleBaseDecode(pack)
	idF32Count := pack.ReadUint16()
	for ;idF32Count > 0; idF32Count-- {
		ackTestSendOk.idF32 = append(ackTestSendOk.idF32, pack.ReadFloat32())
	}
	ackTestSendOk.idOpU8Flag = pack.ReadUint8()
	if ackTestSendOk.idOpU8Flag == 1 {
		ackTestSendOk.idOpU8 = pack.ReadUint8()
	}
	ackTestSendOk.opRoleBaseFlag = pack.ReadUint8()
	if ackTestSendOk.opRoleBaseFlag == 1 {
		ackTestSendOk.opRoleBase = MsgRoleBaseDecode(pack)
	}
	return ackTestSendOk
}

func (this *AckTestSendOk) GetIdU8() uint8 {
	return this.idU8
}

func (this *AckTestSendOk) GetRoleBase() *MsgRoleBase {
	return this.roleBase
}

func (this *AckTestSendOk) GetIdF32() []float32 {
	return this.idF32
}

func (this *AckTestSendOk) GetIdOpU8() uint8 {
	return this.idOpU8
}

func (this *AckTestSendOk) GetOpRoleBase() *MsgRoleBase {
	return this.opRoleBase
}
